### DISCUSSION TITLE
Make "Value Changed" processor dimension sensitive.

### DISCUSSION THREAD

**Discussion by obermeier** [apache/streampipes#1672] [Ideas]
Currently, a value change is detected if the value of a predefined attribute changes between two events.

Since it is possible to set the scope of an attribute as a dimension in the event schema, I like the idea of differentiating the state among these dimensions.
[As it is done in the Datalake sink](https://github.com/apache/streampipes/discussions/1646#discussioncomment-6070481)

What do you think of the idea of storing the state for each dimension (or for a fixed number)? 

E.g.

In this example the attribute source has the property scope dimensional and this three events appear one after the other.

```
	{
          "source": "sensor1",
	  "state": 1
	}

	{ 
          "source": "sensor2",
	  "state": 2
	}
	
	{ 
          "source": "sensor1",
	  "state": 0
	}
```
This modification would result in a single state change when 'sensor1' switches from 1 to 0.

In the existing implementation, two state changes are detected because there is no differentiation between 'sensor1' and 'sensor2'.

**Comment by tenthe** [comment]
Hi, 
I really like the idea for this new feature.
Cheers,
Philipp

**Comment by bossenti** [comment]
see #1673
